_posts/2018-08-01-umbrellas-just-when-it-rains.md

```diff
@@ -0,0 +1,84 @@
+---
+author: Sean Callan
+author_link: https://github.com/doomspork
+categories: general
+date: 2018-08-01
+ayout: post
+title: Umbrellas: only when it rains?
+excerpt: >
+  A look at Umbrella applications and how they can help us write cleaner maintainable code.
+---
+
+The Umbrella application is powerful tool available to us but many people are unfamiliar with them.
+In this post we'll look at Umbrella applications and why we might want to consider using them for our next project.
+
+## Project types
+
+Before we can understand when to use Umbrellas and the role they play, let's review the types of projects available.
+For all intents and purposes there are three project types in the Erlang and Elixir ecosystem: Applications, Libraries, and Umbrellas.
+
+Libraries are packages of code that can be re-used by other projects and applications.
+These projects lack a supervision tree.
+In Elixir a Library would lack the `:mod` key in the Mix `application/0` function.
+
+Applications are like Libraries but contain a supervision tree.
+If you open the `mix.exs` file for a project and see that the `application/0` function defines a `:mod` entry point: you're working with an Application.
+This is a dependency that has, and maintains, state in addition to processes.
+An Application isn't simply a module with simple functions.
+
+Lastly we have Umbrellas applications which are little more than syntactic sugar for managing a collection of Libraries and Applications as a single project.
+Why we might use Umbrellas is what we'll explore here today.
+
+## Getting to know Umbrella Applications
+
+If you haven't already, take a peek at the [Umbrella](/en/lessons/advanced/umbrella-projects/) lesson which provides a great overview of Umbrella applications.
+
+## Separating concerns
+
+In my humble opinion one of the great strengths of Umbrellas is the forced separation of concerns and decoupling of code.
+We have to be explicit with our individual application's configuration and this includes dependencies on other peer applications.
+If our presentation layer doesn't have the internal data application as a dependency there's no chance those two components will become coupled.
+
+While the same separation can be achieved in theory using a singular Application, the Umbrella gives us the added benefit of enforcing it through dependencies.
```

Comment:
Is umbrella truly a proper noun? Idk myself but worth asking I think. Its a first class concept but used to describe systems as a common noun in this context.